x-pack/plugins/data_enhanced/public/search/search_interceptor.ts

```diff
@@ -47,19 +68,29 @@ export class EnhancedSearchInterceptor extends SearchInterceptor {
       : TimeoutErrorMode.CONTACT;
   }
 
-  public search({ id, ...request }: IKibanaSearchRequest, options: IAsyncSearchOptions = {}) {
-    const searchOptions = {
-      strategy: ENHANCED_ES_SEARCH_STRATEGY,
-      ...options,
+  private createRequestHash$(request: IKibanaSearchRequest, options: IAsyncSearchOptions) {
+    const { sessionId, isRestore } = options;
+    const { preference, ...params } = request.params || {};
+    const hashOptions = {
+      ...params,
+      sessionId,
+      isRestore,
     };
-    const { sessionId, strategy, abortSignal } = searchOptions;
-    const search = () => this.runSearch({ id, ...request }, searchOptions);
 
-    const searchAbortController = new SearchAbortController(abortSignal, this.searchTimeout);
-    this.pendingCount$.next(this.pendingCount$.getValue() + 1);
-    const untrackSearch = this.deps.session.isCurrentSession(options.sessionId)
-      ? this.deps.session.trackSearch({ abort: () => searchAbortController.abort() })
-      : undefined;
+    return from(sessionId ? createRequestHash(hashOptions) : of(undefined));
```

Comment:
This comment was skipped https://github.com/elastic/kibana/pull/92439/files?file-filters%5B%5D=.ts&file-filters%5B%5D=.tsx#r601345261

I am still not 100% sure why we rely on `sessionId`. It is still seems to me that this new client-side caching and search-sessions are different features.